cmd/secrets/file_backend.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build secrets
+
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	s "github.com/DataDog/datadog-agent/pkg/secrets"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const (
+	maxSecretFileSize = 8192
+)
+
+type fileBackend struct {
+	rootPath string
+}
+
+func (fb *fileBackend) get(secretID string) s.Secret {
```

Comment:
nit: It's hard to tell that this is an interface-implementation method since it's private. Either we should add a comment to indicate that or make the method public (+ same for the k8s implementation file)

Reply:
Now it's public. The change in https://github.com/DataDog/datadog-agent/pull/9132#discussion_r707661770 needs this to be public.